### DISCUSSION TITLE
PNG output doesn't work

### DISCUSSION THREAD

**Discussion by ferferga** [lowlighter/metrics#1290] [Q&A]
PNG output doesn't work at all.

* Job run with the issue: https://github.com/ferferga/ferferga/actions/runs/3509619152/jobs/5878843255

See outputted files [here](https://github.com/ferferga/ferferga/blob/main/ferferga-metrics_left.png) and [here](https://github.com/ferferga/ferferga/blob/main/ferferga-metrics_right.png)

SVG output was working perfectly (you can try forking my repo and dropping the commit where I introduced PNG), but I prefer to stick with PNG to have a coherent rendering everywhere.

**Comment by lowlighter** [comment]
Seems it's pushing the object rather than saving the buffer 😅 

```json
{"type":"Buffer","data":[137,80,78,71,13,10,26,10,0,0,0,13,73,72,68,82,0,0,1,224,...
```

Will look at it 👍

**Reply by ferferga** [reply]
@lowlighter Even if this is a single hotfix, can you make a new tag and release when this is fixed, please?

**Reply by lowlighter** [reply]
Ok I'll see to make a release a few days after this is patched 👍 

I haven't looked too much into yet, but I think it may have been broken when the JSON output was fixed (it's checking whether the return type is an object rather than an instance of `Buffer`)

But there's a few location to patch within the `app/action/index.mjs` so it may take some time to test properly and ensure it doesn't break the remaining config outputs

**Reply by ferferga** [reply]
@lowlighter I confirm you this is fixed after merging #1295

**Reply by lowlighter** [reply]
Yeah it seems partially fixed

The examples workflow still has the issue though 😕 
[examples/github-metrics.png](https://github.com/lowlighter/metrics/blob/examples/github-metrics.png) has the exact issue while [examples/metrics.example.png](https://github.com/lowlighter/metrics/blob/examples/metrics.example.png) is fine. 
But it may be another issue as the former isn't supposed to be created as all examples are named

I'll try to investigate a bit more on this, but I'll plan a release in the next days (before next week) as you confirmed it seems to be mainly resolved  
Thanks a lot for checking it 👍
